### ISSUE TITLE
`instance.ismethod` incompatible with `pyimport` / JSProxy usage

### ISSUE THREAD

**Issue by JeffersGlass** [pyodide/pyodide#3776]
## 🐛 Bug

Using the `ismethod` function from the `inspect` module doesn't recognize Python methods as methods when:

- The reference to the `inspect` module has been obtained using `pyimport` (2), or 
- Separate JSProxies of the `inspect` module and the method in question are obtained (3)

### To Reproduce

(1) When everything is done in Python, things work fine.
```python
pyodide.runPython(`
    import inspect
    class A():
        def a():
            ...
    instance = A()
    print(f"{inspect.ismethod(instance.a)}")
`);
// True
```
(2) When `pyimport` is used to get a reference to the `inspect` module, `ismethod` reports False:
```python
const instance = pyodide.runPython(`
    class A():
        def a():
            ...
    instance = A()
    instance
`);
const inspectModule = pyodide.pyimport('inspect')
console.log('Is instance a method?', inspectModule.ismethod(instance))
// False
```

(3) When a reference to the inspect module is grabbed from the return value of `runPython`, `ismethod` also reports false.
```python
const instance = pyodide.runPython(`
    class A():
        def a():
            ...
    instance = A()
    instance
`);
const py_ismethod = pyodide.runPython(`
    import inspect
    inspect.ismethod
`)
console.log('Is instance a method?', py_ismethod(instance))
```

### Expected behavior

I would expect `ismethod` to return True in all these situations

### Environment

- Pyodide Version: Tested in 0.22.1 and 0.23.1
- Browser version: Chromium Version 112.0.5615.49 (Official Build) snap (64-bit)

### Additional context

Inspired by the work in [PyScript PR 1240](https://github.com/pyscript/pyscript/pull/1240), which needs to look at whether a Python function is callable and potentially if it's a method.

Similarly, the [Signature](https://docs.python.org/3/library/inspect.html#introspecting-callables-with-the-signature-object) of a bound method reports having one additional parameter (i.e. it seems to be counting 'self') if `inspect.signature` is used in a similar way to (2) or (3) above, while running similar code directly in Python gives the (correct) answer.

**Comment by hoodmane** [comment]
Thanks @JeffersGlass this looks like a very interesting bug report I am looking forward to investigating it.

**Comment by hoodmane** [comment]
Wait should it say
 `inspectModule.ismethod(instance.a)` with an extra `.a`? As written it looks like the expected behavior.

**Comment by hoodmane** [comment]
I looked into this and it seems to me that everything works as expected:
```
pyodide.runPython(`
    import inspect
    class A():
        def a():
            ...
    instance = A()
    print(1, inspect.ismethod(instance)) # False
    print(2, inspect.ismethod(instance.a)) # True
`);
console.log(3, inspectModule.ismethod(instance)); // false
console.log(4, inspectModule.ismethod(instance.a)); // true
```

**Comment by hoodmane** [comment]
Same with signature. I cannot reproduce any unexpected behaviors when using `inspect.signature` from JavaScript.

**Comment by JeffersGlass** [comment]
Well dang, it seems like I was chasing a typo. Thank you for your time @hoodmane .

The original problem in the linked PyScript PR happened when Pyodide was in a worker thread, and referenced via Synclink. There, I was indeed seeing instances where methods with signatures like `someFunc(self, a)` yielded a signature with 2 parameterrs. I had thought I had discovered here that it was a pure Pyodide issue, but it seems I fooled myself.

I'll continue discussion on the actual issue over in that PyScript PR.
